plugins/interactivity-api-quiz-1835fa/src/blocks/quiz-progress-1835fa/render.php

```diff
@@ -26,21 +26,20 @@
 
 <div
 	<?php echo wp_kses_data( get_block_wrapper_attributes() ); ?>
-	data-wp-interactive='{"namespace": "quiz-1835fa-project-store"}'
+	data-wp-interactive="quiz-1835fa-project-store"
 >
 	<div>
 		<strong><?php echo wp_kses_data( __( 'Answered' ) ); ?></strong>: 
-		<span data-wp-text="state.answered"></span> / <span data-wp-text="state.totalQuizzes"></span>
+		<span data-wp-text="state.answered"></span>/<?php echo count( $state['quizzes'] ); ?>
```

Comment:
@luisherranz the portion of code `<?php echo count( $state['quizzes'] );` is throwing an error
I temporarily fixed the error by adding [this](https://github.com/WordPress/block-development-examples/commit/83a34c2297f8c9d0e9c5506bda34cada4ac1bd46#diff-00b4988bac08355bd749f97b72e98026c49a4b7aaeb88c471eab7e830e9e6027R23) and [this](https://github.com/WordPress/block-development-examples/commit/9865bc37df7687b6845b77d8698a8b92c9b7866e#diff-1f1b27451cef00e38158bfa3b48fbdb9fda5d1dcfe528200492091ac3ed01f00R33)

Reply:
`state.quizzes.length` is not dynamically changed in the client, so it should be read from a static PHP variable instead of a client getter.

Apart from that, we need that value in the server so it's properly server-side rendered.

What's the exact error that it throwed in production?

Reply:
By the way, it doesn't work right now because `state.quizzes` is an object, so it should be `Object.keys(state.quizzes).length`.

But, anyway, let's fix it properly 😄

Reply:
This is the error I get with `<?php echo count( $state['quizzes'] ); ?>` on line 33 

```
Answered: /
Warning: Undefined array key "quizzes" in /var/www/html/wp-content/plugins/interactivity-api-quiz-1835fa/build/blocks/quiz-progress-1835fa/render.php on line 33

Fatal error: Uncaught TypeError: count(): Argument #1 ($value) must be of type Countable|array, null given in /var/www/html/wp-content/plugins/interactivity-api-quiz-1835fa/build/blocks/quiz-progress-1835fa/render.php:33 Stack trace: #0 /var/www/html/wp-includes/blocks.php(519): require() #1 /var/www/html/wp-includes/class-wp-block.php(463): {closure}(Array, '', Object(WP_Block)) #2 /var/www/html/wp-includes/blocks.php(1705): WP_Block->render() #3 /var/www/html/wp-includes/blocks.php(1743): render_block(Array) #4 /var/www/html/wp-includes/class-wp-hook.php(324): do_blocks('<!-- wp:block-d...') #5 /var/www/html/wp-includes/plugin.php(205): WP_Hook->apply_filters('<!-- wp:block-d...', Array) #6 /var/www/html/wp-content/plugins/gutenberg/build/block-library/blocks/post-content.php(50): apply_filters('the_content', '<!-- wp:block-d...') #7 /var/www/html/wp-includes/class-wp-block.php(463): gutenberg_render_block_core_post_content(Array, '<!-- wp:block-d...', Object(WP_Block)) #8 /var/www/html/wp-includes/class-wp-block.php(443): WP_Block->render() #9 /var/www/html/wp-includes/blocks.php(1705): WP_Block->render() #10 /var/www/html/wp-includes/blocks.php(1743): render_block(Array) #11 /var/www/html/wp-includes/block-template.php(260): do_blocks('<!-- wp:templat...') #12 /var/www/html/wp-includes/template-canvas.php(12): get_the_block_template_html() #13 /var/www/html/wp-includes/template-loader.php(106): include('/var/www/html/w...') #14 /var/www/html/wp-blog-header.php(19): require_once('/var/www/html/w...') #15 /var/www/html/index.php(17): require('/var/www/html/w...') #16 {main} thrown in /var/www/html/wp-content/plugins/interactivity-api-quiz-1835fa/build/blocks/quiz-progress-1835fa/render.php on line 33
There has been a critical error on this website.

[Learn more about troubleshooting WordPress.](https://wordpress.org/documentation/article/faq-troubleshooting/)
```

Reply:
Oh, wait. I know what's happening. The quiz blocks need to be inner blocks of the quiz progress block. Can you change the blueprint to be like that? I'll make the PR.

Reply:
> The quiz blocks need to be inner blocks of the quiz progress block.

@luisherranz In that case, do we need to set the namespace for the children as it is done [here](https://github.com/WordPress/block-development-examples/pull/122/files#diff-bf00ab6bdc328166e227655df6ec823370ecd0ffb2479b5e0f734dc5c15bd0b5R43)?

Also, are independent blocks unable to share namespace and state/store?

Reply:
> What do you mean?

I mean, can two blocks being at the same level (no inner blocks) in different parts of the HTML share namespace and state/store

For example, having two blocks registered by different plugins, can they share namespace (and store/actions) without a parent/child relationship?

```
<div id="block-registered-by-plugin-1" data-wp-interactive="common-project-namespace">
</div>
```

```
<div id="block-registered-by-plugin-1" data-wp-interactive="common-project-namespace">
</div>
```

Reply:
Oh, yeah, sure 🙂